pages/api/graphql-rest.page.ts

```diff
@@ -1363,32 +1383,21 @@ const cors = Cors({
 
 const apolloServer = new ApolloServer({
   schema,
-  dataSources: () => {
-    return {
-      mpdxRestApi: new MpdxRestApi(),
-    };
-  },
-  context: ({ req }: { req: NextApiRequest }): Partial<Context> => {
-    return { authHeader: req.headers.authorization };
-  },
+  allowBatchedHttpRequests: true,
 });
 
-const startServer = apolloServer.start();
-
-export default cors(async (req, res) => {
+export default cors((req, res) => {
   if (req.method === 'OPTIONS') {
     res.end();
     return false;
   }
-  await startServer;
-  await apolloServer.createHandler({
-    path: '/api/graphql-rest',
+  startServerAndCreateNextHandler(apolloServer, {
```

Comment:
Also, I believe we need to create the handler once outside of this function and then call it inside this function. That should fix the `An error occurred during Apollo Server startup. All GraphQL requests will now fail. The startup error was: You should only call 'start()' or 'startInBackgroundHandlingStartupErrorsByLoggingAndFailingAllRequests()' once on your ApolloServer.` error.


```ts
const handler = startServerAndCreateNextHandler(apolloServer, {
  context: async (req: NextApiRequest): Promise<Context> => ({
    dataSources: {
      mpdxRestApi: new MpdxRestApi(req.headers.authorization ?? ''),
    },
  }),
});

export default cors(async (req, res) => {
  if (req.method === 'OPTIONS') {
    res.end();
    return false;
  }
  await handler(req, res);
});

```